### DISCUSSION TITLE
Is it possible to have different handlebar template layout for different devices?

### DISCUSSION THREAD

**Discussion by arunakshi** [microsoft-search/pnp-modern-search#2382] [Q&A] [answered]
We got a request the following request from client:

1. Desktop version will have search results displayed as document cards in one row (let say the count is 5). Everything can be seen without horizontal scrollbars.  (It is possible) 
2. Mobile version also will have 5 search results displayed in one row but with custom paging (Having dots to represent paging). Due to device size, users will see only the first one and 1/4 of second card immediately, users will see the second card and 1/4 of third card using pagination component. 

> I assume we could achieve custom pagination via custom renderer. The challenge here is the display of results differently in mobile with pagination. 

Could you please let me know, if we can check to apply different template based on different devices?

**Comment by wobba** [comment, marked as answer]
You can add CSS in the templates with device breakpoints. If you use the the debug template in the web part you can see if there is something on the SPFx context you can use, but I didn't spot anything when I checked right now.

If not we could perhaps add a helper to check for the device - similar to https://stackoverflow.com/questions/16990387/handlebar-register-helper-to-do-a-mobile-check

**Comment by arunakshi** [comment]
Thanks alot Wobba for quick reply. 

We tried with implementing custom helper to check if mobile device or not. It is working with custom helper.
